core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -74,7 +74,17 @@ object StaticFile {
       }
     })
 
-  def calcETag[F[_]: Sync]: File => F[String] = f => Sync[F].delay(f.length().toString)
+  def calcETag[F[_]: Sync]: File => F[String] =
+    f =>
+      Sync[F]
+        .delay(f.isFile)
+        .flatMap { isFile =>
+          if (isFile) {
+            Sync[F].delay(s"${f.lastModified()}${f.length()}")
```

Comment:
nginx uses these two components, so we're in good company here.  They render as hex, which I'm not sure is significant.  They also have a `-` delimiter, which is pedantically safer.  (`lastModified` shouldn't ever vary by an order of magnitude, but if it did, a collision could happen.)

Reply:
As I understand, the only think that matters is that there is some change, as long as that is maintained do we need to polish it more?